### PR TITLE
Single value widget

### PR DESCRIPTION
**Pull request by shailee-m**
Thank you for contributing to GeriLife. To maintain the quality of code please make sure of the following before merging a PR.
- [X] Mentioned issues addressed in this PR.
- [X] There are no code-climate issues or have been resolved before the PR is merged.
- [X] Added test cases for new code
- [X] All test cases are passing
- [X] Have used multi-language string literal for all text messages.
- [X] Included code comments

### CODE REVIEW

**Comment by brylie** [comment]
It looks like the panel description texts are swapped:
image

**Comment by brylie** [comment]
Let's go ahead and show five homes in each table along with the metric we are measuring. Is the metric a percentage? If so, perhaps we chould just show it as a number. A horizontal bar chart might also be good here.

File: app/server/methods/activityReports.js [hunk 1]
```diff
@@ -120,3 +129,226 @@ export const mergeHomes = (dataRows, homeIds) => {
...
+          new Date(date).getTime() &&
+        (!resident.moveOut
+          ? true
+          : new Date(resident.moveOut).getTime() >
+            new Date(date).getTime())
+    ),
+  }));
+};
+
+export const enumerateDaysBetweenDates = (startDate, endDate) => {
```
Comment (brylie) [inline app/server/methods/activityReports.js:187]: Remember function docstrings.

File: app/server/methods/activityReports.js [hunk 2]
```diff
@@ -120,3 +129,226 @@ export const mergeHomes = (dataRows, homeIds) => {
...
+          residents,
+          endDate,
+          periodInDays
+        ),
+      },
+    };
+  }, {});
+};
+
+export const getResidentsForEachDay = (
```
Comment (brylie) [inline app/server/methods/activityReports.js:164]: Remember function docstrings.

File: app/server/methods/activityReports.js [hunk 3]
```diff
@@ -120,3 +129,226 @@ export const mergeHomes = (dataRows, homeIds) => {
...
+  const lastDate = moment(endDate).startOf('day');
+
+  while (currDate.add(1, 'days').diff(lastDate) < 0) {
+    dates.push(currDate.clone().toDate());
+  }
+
+  return dates;
+};
+
+export const totalActiveResidentsPerHome = (
```
Comment (brylie) [inline app/server/methods/activityReports.js:200]: Remember function docstrings.

File: app/server/methods/activityReports.js [hunk 4]
```diff
@@ -120,3 +129,226 @@ export const mergeHomes = (dataRows, homeIds) => {
...
+              ) / activityPercentagePerDay.length,
+      };
+    })
+    .sort(
+      (prev, curr) =>
+        curr.averageDailyActivities - prev.averageDailyActivities
+    );
+};
+
+export const getEdgeValues = (homes) => {
```
Comment (brylie) [inline app/server/methods/activityReports.js:315]: Remember function docstrings.

File: app/server/methods/activityReports.js [hunk 5]
```diff
@@ -120,3 +129,226 @@ export const mergeHomes = (dataRows, homeIds) => {
...
+    top5: homes.filter((home) =>
+      top5Values.includes(home.averageDailyActivities)
+    ),
+    bottom5: homes.filter((home) =>
+      bottom5Values.includes(home.averageDailyActivities)
+    ),
+  };
+};
+
+export const getEdgeValuesByPeriod = ({ endDate, periodInDays }) => {
```
Comment (brylie) [inline app/server/methods/activityReports.js:347]: Remember function docstrings.

File: app/server/methods/activityReports.js [hunk 6]
```diff
@@ -120,3 +129,226 @@ export const mergeHomes = (dataRows, homeIds) => {
...
+            moment(date).format('YYYY-MM-DD')
+          );
+        }
+      ),
+    }));
+  });
+  return homeActivityMap;
+};
+
+/**
```
Comment (brylie) [inline app/server/methods/activityReports.js:237]: Add a general description to this docstring along with descriptions of input paramaters.

**Comment by brylie** [comment]
I am  getting the following error:

```sh
(STDERR) Deprecation warning: moment().subtract(period, number) is deprecated. Please use moment().subtract(number, period). See http://momentjs.com/guides/#/warnings/add-inverted-param/ for more info.
W20200708-11:42:11.094(3)? (STDERR) events.js:288
W20200708-11:42:11.094(3)? (STDERR)       throw er; // Unhandled 'error' event
```

**Comment by shailee-m** [comment]
> Let's go ahead and improve the charts so they both always show five homes perhaps? At the very least, the widget should show the metric being measured with a brief description. If it is a percentage value, it may suffice to display the percentage, although a bar chart would allow intuitive comparison.

Fixing to exact 5 values is not a good idea as multiple homes can have the same activity level. I have added the percentage against each home

**Comment by shailee-m** [comment]
> It looks like the panel description texts are swapped:
> image

Yes. Updated the code

**Pull request closed without merging**